docs/source/explanations/dataclass.md

```diff
@@ -0,0 +1,455 @@
+    humidity: float = 70
+    wind: float = 0
+```
+
+## Python `dataclasses.dataclass`
+
+If you're familiar with the builtin python `dataclass` decorator, you will
+immediately recognize the main pattern here.  Indeed, the `evented_dataclass`
+decorator wraps the [`dataclasses.dataclass`
+decorator](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass)
+from the python standard library, and all of the standard arguments to
+`dataclass` apply, along with a few more features relating to events, and
+`properties`.  As such, it is *strongly* recommended that you first review the
+basics in the
+[documentation](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass)
+(or, a more introductory [in-depth
+tutorial](https://realpython.com/python-data-classes/)).
+
+Minimally, you should know that the `@dataclass` decorator auto-generates an
+`__init__` method by examining the class to find `fields`, where a "field" is
+defined as a class variable that has a [type
+annotation](https://docs.python.org/3/glossary.html#term-variable-annotation).
+So you *must* include a type annotation for your field to be included in the
+`__init__` signature.  Variables without a default value are considered
+mandatory
+
+```python
+In [1]: from dataclasses import dataclass
+
+In [2]: @dataclass
+    ...: class Test:
+    ...:     x: int  # mandatory parameter
+    ...:     y: str = 'hi'  # optional parameter
+    ...:     z = 2  # no type annotation, NOT in __init__
+    ...:
+
+In [3]: Test?
+Init signature: Test(x: int, y: str = 'hi') -> None
+Docstring:      Test(x: int, y: str = 'hi')
+```
+
+### Handy `dataclass` features
+
+The python documentation for `dataclass` is good, and relatively short, so we
+won't duplicated it here, but here are some handy features that may be
+particularly useful when defining napari models:
+
+1. Add custom [**post-init
+   processing**](https://docs.python.org/3/library/dataclasses.html#post-init-processing)
+   by declaring a `__post_init__` method on your class.
+2. [**Exclude specific
+   variables**](https://docs.python.org/3/library/dataclasses.html#class-variables)
+   from dataclass fields by annotating as `typing.ClassVar`.
+3. Declare [**init-only
+   variables**](https://docs.python.org/3/library/dataclasses.html#init-only-variables)
+   (variables that are passed to `__post_init__` but are otherwise *not* counted
+   as fields in the dataclass) by annotating as type `dataclasses.InitVar`.
+4. Provide a [**`default_factory`
+   function**](https://docs.python.org/3/library/dataclasses.html#default-factory-functions)
+   to generate the default value for a field using a function (this is important
+   for [mutable default
+   values](https://docs.python.org/3/library/dataclasses.html#mutable-default-values)
+   such as an empty list.)
+    ```python
+    from dataclasses import dataclass, field
+
+    @dataclass
+    class D:
+        x: list = field(default_factory=list)
+    ```
```

Comment:
I havn't had to use this syntax yet (instead using `Property`) and I wonder if we expect it's usage to be so common that we need it in our docs. I don't mind, but I'm with you on trying to keep our docs as minimal as possible

Reply:
which syntax? the default_factory?

Reply:
both `field` and `default_factory`, but maybe I've been using more tuples etc

Reply:
I think it's important enough to mention.  If anyone wants to set an empty list as the default value for something, that's the way to do it... and _not_ `x: list = []`

Reply:
ok, good point

Reply:
I agree that this is good to include! I've come across this myself.